src/include/ctype_inline.h

```diff
@@ -55,7 +55,11 @@ __wt_isdigit(u_char c)
 static inline bool
 __wt_isprint(u_char c)
 {
-    return (isprint(c) != 0);
+    /*
+     * On some systems, isprint() says that characters over 0x80 are printable, even if they may not
+     * actually be printable.
+     */
+    return (isprint(c) != 0) && (c < 0x80);
```

Comment:
I'm slightly wary of this change.  Mac/OS may seem "wrong" in this regard, but it may depend on what locality is set.  On the other hand, I don't see that we use `__wt_isprint` to decide whether table names or valid, where we might break international programs.  So it seems safe.  So, I'm curious the call path that got us to isprint in this bug?

Reply:
It started with `__rts_btree_abort_ondisk_kv()` on the history store, which then prints a log message using this code:

```
    __wt_verbose_level_multi(session, WT_VERB_RECOVERY_RTS(session), WT_VERBOSE_DEBUG_2,
      WT_RTS_VERB_TAG_ONDISK_KV_REMOVE "removing the key, tombstone=%s, key=%s",
      upd->type == WT_UPDATE_TOMBSTONE ? "true" : "false",
      __wt_key_string(session, key->data, key->size, S2BT(session)->key_format, key_string));
```

Then `__wt_key_string()` calls `__wt_buf_set_printable_format()`, which then calls `__wt_buf_set_printable()` for the record key byte-string part of the History Store key. And within it, `__wt_raw_to_esc_hex()` uses `__wt_isprint()` to determine whether to print the character directly or print it as an escaped hex value.

The locale is set to `en_US.UTF-8`, but I guess it isn't `C`, which perhaps the rest of the code implicitly expects? The current behavior is clearly not what `Py_BuildValue()` expects, which we call later in our Python wrapper code that then ultimately implements the message handler.

Reply:
Thanks for the explanation!

Reply:
I'm satisfied that we're using `__wt_isascii` solely for the purpose of seeing if a character can be printed in an error message, or printing contents of a buffer.  We never use it to decide if a symbol in a namespace is legal (like session.create names).